azurelinuxagent/common/conf.py

```diff
@@ -707,3 +707,11 @@ def get_log_collector_initial_delay(conf=__conf__):
     NOTE: This option is experimental and may be removed in later versions of the Agent.
     """
     return conf.get_int("Debug.LogCollectorInitialDelay", 5 * 60)
+
+def get_disable_rsm_downgrade(conf=__conf__):
+    """
+    If True, the agent will not downgrade to a lower version when a lower version is requested in the goal state.
+
+    Todo: Flag will be removed once we have a fix for rsm downgrade scenario.
+    """
+    return conf.get_switch("Debug.DisableRsmDowngrade", True)
```

Comment:
Had to add new flag in order to validate rsm upgrade scenario in e2e tests. Unfortunately, we have to downgrade to lower version first to test upgrade.

Reply:
Most options allow enable instead of disable. Should we change this to get_enable_rsm_downgrade with False default